app/routes/products_.$productId/route.tsx

```diff
@@ -31,19 +31,36 @@ export const loader = async ({ params, request }: LoaderFunctionArgs) => {
 export default function ProductDetailsPage() {
     const { product } = useLoaderData<typeof loader>();
     const { setIsOpen } = useCartOpen();
+    const [addToCartAttempted, setAddToCartAttempted] = useState(false);
 
     const { trigger: addToCart } = useAddToCart();
     const quantityInput = useRef<HTMLInputElement>(null);
 
-    const [selectedOptions, setSelectedOptions] = useState<Record<string, string>>({});
+    const [selectedOptions, setSelectedOptions] = useState<Record<string, string | null>>(
+        product.productOptions
```

Comment:
it's so complicated?
```
const getInitialSelectedOptions = (productOptions) => {
   const result = {};
   for (const option of options) {
      if (option.name) result[option.name] = null;
   }
}
```

Reply:
yes, this is a little bit complicated because of crazy types from sdk, but please check out updated version